### DISCUSSION TITLE
Abandoned?

### DISCUSSION THREAD

**Discussion by Xyndra** [jni-rs/jni-rs#546] [Q&A]
This is a really cool project, but is it abandoned? From what I see, everything here is from many months ago, but I don't know of any alternatives

**Comment by argv-minus-one** [comment]
@rib is the maintainer. He seems to have been mostly working on other projects this year, but he did open https://github.com/jni-rs/jni-sys/pull/37 two days ago.

**Comment by rib** [comment]
Hi, yes unfortunately I've not had time in the last year to actively maintain jni-rs. I've been sort of hoping to get back to it and get a chance to at least wrap up a release but it doesn't seem to be happening.

@argv-minus-one I wonder if you'd be happy to take over maintenance? I've interacted with you via a lot of PRs for the project and trust that you have a good familiarity with the fiddly details of JNI and are also familiar with many of the changes that were landing last year.

**Reply by argv-minus-one** [reply]
I don't have the bandwidth either. There was a time when I'd have been happy to, but that's no longer the case. 😞

**Reply by rib** [reply]
Ah, 😞

As it happens the contract that has been keeping me busy this last year is coming to an end in a few weeks and so there's a chance I can try and spend a small amount of time looking at jni-rs

**Comment by ssrlive** [comment]
Hi @rib, I have make a PR #559 to fit the new rust compiler. Please review it and/or merge it. And I'm interested in maintaining this crate, you can add me as a maintainer and publisher.

**Comment by rib** [comment]
Although there was a long gap between major releases, there has been quite a lot of work put into the project on and off over the years and I've just published a (long overdue) jni 0.22 release: https://github.com/jni-rs/jni-rs/pull/650.

I can't promise that I'll always have bandwidth to be very actively working on this crate but I wouldn't say that the project is abandoned.

If anyone is interested in helping maintain the project then the best place to start would be via opening PRs that address various issues and contributing to discussions showing a good familiarity with JNI over time.

I'll close this for now.
